app/components/AddPasswordToKeychainModal/AddPasswordToKeychainModal.component.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import type { FC } from 'react';
+
+import {
+  Box,
+  Button,
+  createStyles,
+  FormHelperText,
+  makeStyles,
+  Modal,
+  Typography,
+} from '@material-ui/core';
+import { Field, Form, Formik } from 'formik';
+import { TextField } from 'formik-material-ui';
+import * as Yup from 'yup';
+
+import type { Theme } from '../../theme';
+import { MOBIcon } from '../icons';
+
+export interface AddPasswordToKeychainModalProps {
+  open: boolean;
+  passwordLength: number;
+}
```

Comment:
I would prefer this in a .d.ts file

Reply:
Hmmm, I'm unfamiliar with that, could you elaborate?